File: Baekjoon/seongjae/5_week/10828.cpp
#include <iostream>
#include <stack>

using namespace std;

int main() {
    int n; // 명령 횟수
    cin >> n;
    stack<int> stack; // 스택

    string comm; // 명령어 변수
    int pushNum = 0; // push 정수

    for (int i = 0; i < n; i++) {
        cin >> comm; // 명령어
        if (comm == "pop") { // pop
            if (stack.empty()) {
                cout << -1 << endl;
            } else {
                cout << stack.top() << endl;
                stack.pop();
            }
        } else if (comm == "size") { // size
            cout << stack.size() << endl;
        } else if (comm == "empty") { // empty
            cout << stack.empty() << endl;
        } else if (comm == "top") { // top
            if (stack.empty()) {
                cout << -1 << endl;
            } else cout << stack.top() << endl;
        } else { // push
            cin >> pushNum; // 정수 X 읽기
            stack.push(pushNum);
        }
    }
}

File: Baekjoon/heemang/5_week/10828.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        Stack<String> stack = new Stack<>();
        StringBuilder sb = new StringBuilder();
        while (n-- > 0) {
            String input = br.readLine();

            if (input.startsWith("push")) {
                stack.push(input.split(" ")[1]);
                continue;
            }

            if (stack.isEmpty()) {
                switch (input) {
                    case "pop":
                    case "top":
                        sb.append(-1).append("\n");
                        break;
                    case "size":
                        sb.append(0).append("\n");
                        break;
                    default:
                        sb.append(1).append("\n");
                }
                continue;
            }

            switch (input) {
                case "pop":
                    sb.append(stack.pop()).append("\n");
                    break;
                case "size":
                    sb.append(stack.size()).append("\n");
                    break;
                case "empty":
                    sb.append(0).append("\n");
                    break;
                default:
                    sb.append(stack.peek()).append("\n");
            }
        }

        System.out.print(sb);
    }
}
